ambry-api/src/main/java/com.github.ambry/router/KeyManagementService.java

```diff
@@ -37,6 +37,13 @@
    */
   void register(short accountId, short containerId) throws GeneralSecurityException;
 
+  /**
+   * Registers with KMS to create key for a unique context.
+   * @param context refers to the key context to register
+   * @throws {@link GeneralSecurityException} on KMS unavailability or duplicate registration
+   */
+  void register(String context) throws GeneralSecurityException;
```

Comment:
What would be the context for cloudBlobStore ?

Reply:
The name of the Azure storage account, eg. wus2ambryprodblobstore1, or a more friendly name like "AmbryAzureProd".  This would replace the container-specific context "accountId:containerId" for blobs in containers with no registered key.